src/test/java/JiraTesterBearerAuth.java

```diff
@@ -0,0 +1,123 @@
+
+        final String projectKey = "TESTPROJECT";
+        final String issueId = "TESTPROJECT-425";
+        final Integer actionId = 21;
+
+        final Issue issue = restService.getIssue(issueId);
+        System.out.println("issue:" + issue);
+
+
+        final List<Transition> availableActions = restService.getAvailableActions(issueId);
+        for (Transition action : availableActions) {
+            System.out.println("Action:" + action);
+        }
+
+        for (IssueType issueType : restService.getIssueTypes()) {
+            System.out.println(" issue type: " + issueType);
+        }
+
+//        restService.addVersion("TESTPROJECT", "0.0.2");
+
+        final List<Component> components = restService.getComponents(projectKey);
+        for (Component component : components) {
+            System.out.println("component: " + component);
+        }
+
+//        BasicComponent backendComponent = null;
+//        final Iterable<BasicComponent> components1 = Lists.newArrayList(backendComponent);
+//        restService.createIssue("TESTPROJECT", "This is a test issue created using Jira jenkins plugin. Please ignore it.", "TESTUSER", components1, "test issue from Jira jenkins plugin");
+
+        final List<Issue> searchResults = restService.getIssuesFromJqlSearch("project = \"TESTPROJECT\"", 3);
+        for (Issue searchResult : searchResults) {
+            System.out.println("JQL search result: " + searchResult);
+        }
+
+        final List<String> projectsKeys = restService.getProjectsKeys();
+        for (String projectsKey : projectsKeys) {
+            System.out.println("project key: " + projectsKey);
+        }
+
+        final List<Status> statuses = restService.getStatuses();
+        for (Status status : statuses) {
+            System.out.println("status:" + status);
+        }
+
+        final User user = restService.getUser("TESTUSER");
+        System.out.println("user: " + user);
+
+        final List<ExtendedVersion> versions = restService.getVersions(projectKey);
+        for (ExtendedVersion version : versions) {
+            System.out.println("version: "  + version);
+        }
+
+//        Version releaseVersion = new Version(version.getSelf(), version.getId(), version.getName(),
+//                version.getDescription(), version.isArchived(), true, new DateTime());
+//        System.out.println(" >>>> release version 0.0.2");
+//        restService.releaseVersion("TESTPROJECT", releaseVersion);
+
+//        System.out.println(" >>> update issue TESTPROJECT-425");
+//        restService.updateIssue(issueId, Collections.singletonList(releaseVersion));
+
+//        final Issue updatedIssue = restService.progressWorkflowAction(issueId, actionId);
+//        System.out.println("Updated issue:" + updatedIssue);
+
+
+
+        for(int i=0;i<10;i++){
+            callUniq( restService );
```

Comment:
Why is the loop needed / what is that test case trying to achieve? I would prefer this one big method to be splitted into smaller test cases with relevant assertions because otherwise it's hard to say what you're testing - see other [test cases](https://github.com/jenkinsci/jira-plugin/blob/master/src/test/java/hudson/plugins/jira/JiraSiteTest.java)

Reply:
Thank you very much for your review and your suggestions!! 🙏  What I did is replicate code inside [JiraTester](https://github.com/jenkinsci/jira-plugin/blob/master/src/test/java/JiraTester.java) class. Do you think that add another class similar to [JiraRestServiceTest](https://github.com/jenkinsci/jira-plugin/blob/master/src/test/java/hudson/plugins/jira/JiraRestServiceTest.java) is enough? 🤔  New method `buildBaseApiPath` is implicit tested by already existing test methods inside `JiraRestServiceTest` class

Reply:
Ah I forgot about this _manual_ test class created long time ago ;-) 
Take a look at [JiraSiteTest](https://github.com/jenkinsci/jira-plugin/blob/master/src/test/java/hudson/plugins/jira/JiraSiteTest.java), I think that after updating the UI, adapting this Test class should be enough here.

Reply:
Do you think that token should be passed as username and password or secret text?